web/src/main/webapp/my-app/layout/controllers.js

```diff
@@ -366,6 +370,23 @@ define(['angular', 'jquery'], function(angular, $) {
             $sessionStorage.layout = layoutService.getLayout();
           });
       };
+      vm.setGuestMode = function() {
+        $scope.guestMode = false;
+        layoutService.getGuestMode().then(function(result) {
+          if (angular.isDefined(result) && !result) {
```

Comment:
Would this be more readable as an `if...else if` sequence? something like

```
if (result) {
  $scope.guestMode = true;
   return true;
} else if (angular.isDefined(result)) {
  $scope.guestMode = false;
  return false;
} else {
  return result;
}
```

How about reducing the number of places this function returns, would this be an improvement?

```
if (result) {
  $scope.guestMode = true;
} else if (angular.isDefined(result)) {
  $scope.guestMode = false;
} 

return result;
```

Or what about reducing the number of conditions further? Since it defaults `$scope.guestMode = false` presumably it doesn't need to do anything unless `$scope.guestMode` instead needs to be set to true?

```
vm.setGuestMode = function() {
  $scope.guestMode = false;

  layoutService.getGuestMode().then(function(result) {
    if (result) {
      $scope.guestMode = true;
    }
    return result;
  })
  .catch(function() {
    $log.warn('could not retrieve guest mode');
   });
};
```